CI: TestControlPlane/Services/NodePort/v1.25/nodeport-worker2

```
--- FAIL: TestControlPlane (20.54s)
                                   
    --- FAIL: TestControlPlane/Services/NodePort (11.46s)
                                                         
        --- FAIL: TestControlPlane/Services/NodePort/v1.25/nodeport-worker2 (10.12s)
                                                                                    
level=warning msg="Running Cilium with \"kvstore\"=\"\" requires identity allocation via CRDs. Changing identity-allocation-mode to \"crd\"" subsys=config
level=info msg="Trying to auto-enable \"enable-node-port\", \"enable-external-ips\", \"bpf-lb-sock\", \"enable-host-port\", \"enable-session-affinity\" features" subsys=daemon
level=warning msg="Unable to automatically detect MTU" error="No interface contains the provided ip: 172.18.0.3" subsys=mtu
level=info msg="L7 proxies are disabled" subsys=daemon
level=warning msg="No valid cgroup base path found: socket load-balancing tracing feature will not work. File a GitHub issuewith an example cgroup path for a pod by running command on Kubernetes node: sudo crictl inspectp -o=json $POD_ID | grep cgroupsPath" subsys=cgroup-manager
level=info msg="Reading old endpoints..." subsys=daemon
level=info msg="No old endpoints found." subsys=daemon
level=info msg="Retrieved node information from kubernetes node" nodeName=nodeport-worker2 subsys=k8s
level=info msg="Received own node information from API server" ipAddr.ipv4=172.18.0.5 ipAddr.ipv6="fc00:f853:ccd:e793::5" k8sNodeIP=172.18.0.5 labels="map[beta.kubernetes.io/arch:amd64 beta.kubernetes.io/os:linux cilium.io/ci-node:k8s2 kubernetes.io/arch:amd64 kubernetes.io/hostname:nodeport-worker2 kubernetes.io/os:linux]" nodeName=nodeport-worker2 subsys=k8s v4Prefix=10.244.2.0/24 v6Prefix="fd00:10:244:2::/64"
level=info msg="k8s mode: Allowing localhost to reach local endpoints" subsys=daemon
level=info msg="Detected devices" devices="[]" subsys=linux-datapath
level=info msg="Enabling k8s event listener" subsys=k8s-watcher
level=info msg="Removing stale endpoint interfaces" subsys=daemon
level=info msg="Skipping kvstore configuration" subsys=daemon
level=info msg="Initializing node addressing" subsys=daemon
level=info msg="Initializing kubernetes IPAM" subsys=ipam v4Prefix=1.1.1.0/24 v6Prefix="cafe::/96"
level=info msg="Restoring endpoints..." subsys=daemon
level=info msg="Endpoints restored" failed=0 restored=0 subsys=daemon
level=info msg="Addressing information:" subsys=daemon
level=info msg="  Cluster-Name: default" subsys=daemon
level=info msg="  Cluster-ID: 0" subsys=daemon
level=info msg="  Local node-name: nodeport-worker2" subsys=daemon
level=info msg="  Node-IPv6: fc00:f853:ccd:e793::5" subsys=daemon
level=info msg="  External-Node IPv4: 172.18.0.5" subsys=daemon
level=info msg="  Internal-Node IPv4: 1.1.1.2" subsys=daemon
level=info msg="  IPv4 allocation prefix: 10.244.2.0/24" subsys=daemon
level=info msg="  Loopback IPv4: 169.254.42.1" subsys=daemon
level=info msg="  Local IPv4 addresses:" subsys=daemon
level=info msg="  - 10.0.0.2" subsys=daemon
level=info msg="  - 10.0.0.3" subsys=daemon
level=info msg="  - 10.0.0.4" subsys=daemon
level=info msg="Creating or updating CiliumNode resource" node=nodeport-worker2 subsys=nodediscovery
level=info msg="Initializing identity allocator" subsys=identity-cache
level=info msg="Allocating identities between range" cluster-id=0 max=65535 min=256 subsys=identity-cache
level=info msg="Waiting until local node addressing before starting watchers depending on it" subsys=k8s-watcher
level=info msg="Waiting until all pre-existing resources have been received" subsys=k8s-watcher
level=info msg="Adding local node to cluster" node="{nodeport-worker2 default [{InternalIP 172.18.0.5} {InternalIP fc00:f853:ccd:e793::5} {CiliumInternalIP 1.1.1.2} {CiliumInternalIP cafe::2}] 10.244.2.0/24 [] fd00:10:244:1::/64 [] 1.1.1.207 cafe::89b <nil> <nil> 0 local 0 map[beta.kubernetes.io/arch:amd64 beta.kubernetes.io/os:linux cilium.io/ci-node:k8s2 kubernetes.io/arch:amd64 kubernetes.io/hostname:nodeport-worker2 kubernetes.io/os:linux] 1 }" subsys=nodediscovery
level=warning msg="UpdateIdentities: Updating an existing identity" identity=16777217 labels="[reserved:kube-apiserver]" labels(new)="[cidr:0.0.0.0/0 cidr:128.0.0.0/1 cidr:128.0.0.0/2 cidr:160.0.0.0/3 cidr:160.0.0.0/4 cidr:168.0.0.0/5 cidr:172.0.0.0/10 cidr:172.0.0.0/11 cidr:172.0.0.0/6 cidr:172.0.0.0/7 cidr:172.0.0.0/8 cidr:172.0.0.0/9 cidr:172.16.0.0/12 cidr:172.16.0.0/13 cidr:172.16.0.0/14 cidr:172.18.0.0/15 cidr:172.18.0.0/16 cidr:172.18.0.0/17 cidr:172.18.0.0/18 cidr:172.18.0.0/19 cidr:172.18.0.0/20 cidr:172.18.0.0/21 cidr:172.18.0.0/22 cidr:172.18.0.0/23 cidr:172.18.0.0/24 cidr:172.18.0.0/25 cidr:172.18.0.0/26 cidr:172.18.0.0/27 cidr:172.18.0.0/28 cidr:172.18.0.0/29 cidr:172.18.0.0/30 cidr:172.18.0.2/31 cidr:172.18.0.3/32 reserved:kube-apiserver reserved:world]" subsys=policy
level=warning msg="UpdateIdentities: Updating an existing identity" identity=16777217 labels="[reserved:kube-apiserver]" labels(new)="[cidr:0.0.0.0/0 cidr:128.0.0.0/1 cidr:128.0.0.0/2 cidr:160.0.0.0/3 cidr:160.0.0.0/4 cidr:168.0.0.0/5 cidr:172.0.0.0/10 cidr:172.0.0.0/11 cidr:172.0.0.0/6 cidr:172.0.0.0/7 cidr:172.0.0.0/8 cidr:172.0.0.0/9 cidr:172.16.0.0/12 cidr:172.16.0.0/13 cidr:172.16.0.0/14 cidr:172.18.0.0/15 cidr:172.18.0.0/16 cidr:172.18.0.0/17 cidr:172.18.0.0/18 cidr:172.18.0.0/19 cidr:172.18.0.0/20 cidr:172.18.0.0/21 cidr:172.18.0.0/22 cidr:172.18.0.0/23 cidr:172.18.0.0/24 cidr:172.18.0.0/25 cidr:172.18.0.0/26 cidr:172.18.0.0/27 cidr:172.18.0.0/28 cidr:172.18.0.0/29 cidr:172.18.0.0/30 cidr:172.18.0.2/31 cidr:172.18.0.3/32 reserved:kube-apiserver reserved:world]" subsys=policy
level=warning msg="UpdateIdentities: Updating an existing identity" identity=16777217 labels="[reserved:kube-apiserver]" labels(new)="[cidr:0.0.0.0/0 cidr:128.0.0.0/1 cidr:128.0.0.0/2 cidr:160.0.0.0/3 cidr:160.0.0.0/4 cidr:168.0.0.0/5 cidr:172.0.0.0/10 cidr:172.0.0.0/11 cidr:172.0.0.0/6 cidr:172.0.0.0/7 cidr:172.0.0.0/8 cidr:172.0.0.0/9 cidr:172.16.0.0/12 cidr:172.16.0.0/13 cidr:172.16.0.0/14 cidr:172.18.0.0/15 cidr:172.18.0.0/16 cidr:172.18.0.0/17 cidr:172.18.0.0/18 cidr:172.18.0.0/19 cidr:172.18.0.0/20 cidr:172.18.0.0/21 cidr:172.18.0.0/22 cidr:172.18.0.0/23 cidr:172.18.0.0/24 cidr:172.18.0.0/25 cidr:172.18.0.0/26 cidr:172.18.0.0/27 cidr:172.18.0.0/28 cidr:172.18.0.0/29 cidr:172.18.0.0/30 cidr:172.18.0.2/31 cidr:172.18.0.3/32 reserved:kube-apiserver reserved:world]" subsys=policy
level=warning msg="UpdateIdentities: Updating an existing identity" identity=16777217 labels="[reserved:kube-apiserver]" labels(new)="[cidr:0.0.0.0/0 cidr:128.0.0.0/1 cidr:128.0.0.0/2 cidr:160.0.0.0/3 cidr:160.0.0.0/4 cidr:168.0.0.0/5 cidr:172.0.0.0/10 cidr:172.0.0.0/11 cidr:172.0.0.0/6 cidr:172.0.0.0/7 cidr:172.0.0.0/8 cidr:172.0.0.0/9 cidr:172.16.0.0/12 cidr:172.16.0.0/13 cidr:172.16.0.0/14 cidr:172.18.0.0/15 cidr:172.18.0.0/16 cidr:172.18.0.0/17 cidr:172.18.0.0/18 cidr:172.18.0.0/19 cidr:172.18.0.0/20 cidr:172.18.0.0/21 cidr:172.18.0.0/22 cidr:172.18.0.0/23 cidr:172.18.0.0/24 cidr:172.18.0.0/25 cidr:172.18.0.0/26 cidr:172.18.0.0/27 cidr:172.18.0.0/28 cidr:172.18.0.0/29 cidr:172.18.0.4/30 cidr:172.18.0.4/31 cidr:172.18.0.4/32 reserved:kube-apiserver reserved:world]" subsys=policy
level=warning msg="UpdateIdentities: Updating an existing identity" identity=16777217 labels="[reserved:kube-apiserver]" labels(new)="[cidr:0.0.0.0/0 cidr:128.0.0.0/1 cidr:128.0.0.0/2 cidr:160.0.0.0/3 cidr:160.0.0.0/4 cidr:168.0.0.0/5 cidr:172.0.0.0/10 cidr:172.0.0.0/11 cidr:172.0.0.0/6 cidr:172.0.0.0/7 cidr:172.0.0.0/8 cidr:172.0.0.0/9 cidr:172.16.0.0/12 cidr:172.16.0.0/13 cidr:172.16.0.0/14 cidr:172.18.0.0/15 cidr:172.18.0.0/16 cidr:172.18.0.0/17 cidr:172.18.0.0/18 cidr:172.18.0.0/19 cidr:172.18.0.0/20 cidr:172.18.0.0/21 cidr:172.18.0.0/22 cidr:172.18.0.0/23 cidr:172.18.0.0/24 cidr:172.18.0.0/25 cidr:172.18.0.0/26 cidr:172.18.0.0/27 cidr:172.18.0.0/28 cidr:172.18.0.0/29 cidr:172.18.0.4/30 cidr:172.18.0.4/31 cidr:172.18.0.4/32 reserved:kube-apiserver reserved:world]" subsys=policy
level=warning msg="UpdateIdentities: Updating an existing identity" identity=16777217 labels="[reserved:kube-apiserver]" labels(new)="[cidr:0.0.0.0/0 cidr:128.0.0.0/1 cidr:128.0.0.0/2 cidr:160.0.0.0/3 cidr:160.0.0.0/4 cidr:168.0.0.0/5 cidr:172.0.0.0/10 cidr:172.0.0.0/11 cidr:172.0.0.0/6 cidr:172.0.0.0/7 cidr:172.0.0.0/8 cidr:172.0.0.0/9 cidr:172.16.0.0/12 cidr:172.16.0.0/13 cidr:172.16.0.0/14 cidr:172.18.0.0/15 cidr:172.18.0.0/16 cidr:172.18.0.0/17 cidr:172.18.0.0/18 cidr:172.18.0.0/19 cidr:172.18.0.0/20 cidr:172.18.0.0/21 cidr:172.18.0.0/22 cidr:172.18.0.0/23 cidr:172.18.0.0/24 cidr:172.18.0.0/25 cidr:172.18.0.0/26 cidr:172.18.0.0/27 cidr:172.18.0.0/28 cidr:172.18.0.0/29 cidr:172.18.0.4/30 cidr:172.18.0.4/31 cidr:172.18.0.4/32 reserved:kube-apiserver reserved:world]" subsys=policy
level=warning msg="UpdateIdentities: Updating an existing identity" identity=16777217 labels="[reserved:kube-apiserver]" labels(new)="[cidr:0.0.0.0/0 cidr:128.0.0.0/1 cidr:128.0.0.0/2 cidr:160.0.0.0/3 cidr:160.0.0.0/4 cidr:168.0.0.0/5 cidr:172.0.0.0/10 cidr:172.0.0.0/11 cidr:172.0.0.0/6 cidr:172.0.0.0/7 cidr:172.0.0.0/8 cidr:172.0.0.0/9 cidr:172.16.0.0/12 cidr:172.16.0.0/13 cidr:172.16.0.0/14 cidr:172.18.0.0/15 cidr:172.18.0.0/16 cidr:172.18.0.0/17 cidr:172.18.0.0/18 cidr:172.18.0.0/19 cidr:172.18.0.0/20 cidr:172.18.0.0/21 cidr:172.18.0.0/22 cidr:172.18.0.0/23 cidr:172.18.0.0/24 cidr:172.18.0.0/25 cidr:172.18.0.0/26 cidr:172.18.0.0/27 cidr:172.18.0.0/28 cidr:172.18.0.0/29 cidr:172.18.0.4/30 cidr:172.18.0.4/31 cidr:172.18.0.4/32 reserved:kube-apiserver reserved:world]" subsys=policy
level=info msg="regenerating all endpoints" reason="one or more identities created or deleted" subsys=endpoint-manager
level=warning msg="UpdateIdentities: Updating an existing identity" identity=16777217 labels="[reserved:kube-apiserver]" labels(new)="[cidr:0.0.0.0/0 cidr:128.0.0.0/1 cidr:128.0.0.0/2 cidr:160.0.0.0/3 cidr:160.0.0.0/4 cidr:168.0.0.0/5 cidr:172.0.0.0/10 cidr:172.0.0.0/11 cidr:172.0.0.0/6 cidr:172.0.0.0/7 cidr:172.0.0.0/8 cidr:172.0.0.0/9 cidr:172.16.0.0/12 cidr:172.16.0.0/13 cidr:172.16.0.0/14 cidr:172.18.0.0/15 cidr:172.18.0.0/16 cidr:172.18.0.0/17 cidr:172.18.0.0/18 cidr:172.18.0.0/19 cidr:172.18.0.0/20 cidr:172.18.0.0/21 cidr:172.18.0.0/22 cidr:172.18.0.0/23 cidr:172.18.0.0/24 cidr:172.18.0.0/25 cidr:172.18.0.0/26 cidr:172.18.0.0/27 cidr:172.18.0.0/28 cidr:172.18.0.0/29 cidr:172.18.0.0/30 cidr:172.18.0.2/31 cidr:172.18.0.3/32 reserved:kube-apiserver reserved:world]" subsys=policy
level=warning msg="UpdateIdentities: Updating an existing identity" identity=16777217 labels="[reserved:kube-apiserver]" labels(new)="[cidr:0.0.0.0/0 cidr:128.0.0.0/1 cidr:128.0.0.0/2 cidr:160.0.0.0/3 cidr:160.0.0.0/4 cidr:168.0.0.0/5 cidr:172.0.0.0/10 cidr:172.0.0.0/11 cidr:172.0.0.0/6 cidr:172.0.0.0/7 cidr:172.0.0.0/8 cidr:172.0.0.0/9 cidr:172.16.0.0/12 cidr:172.16.0.0/13 cidr:172.16.0.0/14 cidr:172.18.0.0/15 cidr:172.18.0.0/16 cidr:172.18.0.0/17 cidr:172.18.0.0/18 cidr:172.18.0.0/19 cidr:172.18.0.0/20 cidr:172.18.0.0/21 cidr:172.18.0.0/22 cidr:172.18.0.0/23 cidr:172.18.0.0/24 cidr:172.18.0.0/25 cidr:172.18.0.0/26 cidr:172.18.0.0/27 cidr:172.18.0.0/28 cidr:172.18.0.0/29 cidr:172.18.0.4/30 cidr:172.18.0.4/31 cidr:172.18.0.4/32 reserved:kube-apiserver reserved:world]" subsys=policy
level=warning msg="UpdateIdentities: Updating an existing identity" identity=16777217 labels="[reserved:kube-apiserver]" labels(new)="[cidr:0.0.0.0/0 cidr:128.0.0.0/1 cidr:128.0.0.0/2 cidr:160.0.0.0/3 cidr:160.0.0.0/4 cidr:168.0.0.0/5 cidr:172.0.0.0/10 cidr:172.0.0.0/11 cidr:172.0.0.0/6 cidr:172.0.0.0/7 cidr:172.0.0.0/8 cidr:172.0.0.0/9 cidr:172.16.0.0/12 cidr:172.16.0.0/13 cidr:172.16.0.0/14 cidr:172.18.0.0/15 cidr:172.18.0.0/16 cidr:172.18.0.0/17 cidr:172.18.0.0/18 cidr:172.18.0.0/19 cidr:172.18.0.0/20 cidr:172.18.0.0/21 cidr:172.18.0.0/22 cidr:172.18.0.0/23 cidr:172.18.0.0/24 cidr:172.18.0.0/25 cidr:172.18.0.0/26 cidr:172.18.0.0/27 cidr:172.18.0.0/28 cidr:172.18.0.0/29 cidr:172.18.0.4/30 cidr:172.18.0.4/31 cidr:172.18.0.4/32 reserved:kube-apiserver reserved:world]" subsys=policy
level=warning msg="UpdateIdentities: Updating an existing identity" identity=16777217 labels="[reserved:kube-apiserver]" labels(new)="[cidr:0.0.0.0/0 cidr:128.0.0.0/1 cidr:128.0.0.0/2 cidr:160.0.0.0/3 cidr:160.0.0.0/4 cidr:168.0.0.0/5 cidr:172.0.0.0/10 cidr:172.0.0.0/11 cidr:172.0.0.0/6 cidr:172.0.0.0/7 cidr:172.0.0.0/8 cidr:172.0.0.0/9 cidr:172.16.0.0/12 cidr:172.16.0.0/13 cidr:172.16.0.0/14 cidr:172.18.0.0/15 cidr:172.18.0.0/16 cidr:172.18.0.0/17 cidr:172.18.0.0/18 cidr:172.18.0.0/19 cidr:172.18.0.0/20 cidr:172.18.0.0/21 cidr:172.18.0.0/22 cidr:172.18.0.0/23 cidr:172.18.0.0/24 cidr:172.18.0.0/25 cidr:172.18.0.0/26 cidr:172.18.0.0/27 cidr:172.18.0.0/28 cidr:172.18.0.0/29 cidr:172.18.0.4/30 cidr:172.18.0.4/31 cidr:172.18.0.4/32 reserved:kube-apiserver reserved:world]" subsys=policy
level=warning msg="UpdateIdentities: Updating an existing identity" identity=16777217 labels="[reserved:kube-apiserver]" labels(new)="[cidr:0.0.0.0/0 cidr:128.0.0.0/1 cidr:128.0.0.0/2 cidr:160.0.0.0/3 cidr:160.0.0.0/4 cidr:168.0.0.0/5 cidr:172.0.0.0/10 cidr:172.0.0.0/11 cidr:172.0.0.0/6 cidr:172.0.0.0/7 cidr:172.0.0.0/8 cidr:172.0.0.0/9 cidr:172.16.0.0/12 cidr:172.16.0.0/13 cidr:172.16.0.0/14 cidr:172.18.0.0/15 cidr:172.18.0.0/16 cidr:172.18.0.0/17 cidr:172.18.0.0/18 cidr:172.18.0.0/19 cidr:172.18.0.0/20 cidr:172.18.0.0/21 cidr:172.18.0.0/22 cidr:172.18.0.0/23 cidr:172.18.0.0/24 cidr:172.18.0.0/25 cidr:172.18.0.0/26 cidr:172.18.0.0/27 cidr:172.18.0.0/28 cidr:172.18.0.0/29 cidr:172.18.0.4/30 cidr:172.18.0.4/31 cidr:172.18.0.4/32 reserved:kube-apiserver reserved:world]" subsys=policy
level=warning msg="UpdateIdentities: Updating an existing identity" identity=16777217 labels="[reserved:kube-apiserver]" labels(new)="[cidr:0.0.0.0/0 cidr:128.0.0.0/1 cidr:128.0.0.0/2 cidr:160.0.0.0/3 cidr:160.0.0.0/4 cidr:168.0.0.0/5 cidr:172.0.0.0/10 cidr:172.0.0.0/11 cidr:172.0.0.0/6 cidr:172.0.0.0/7 cidr:172.0.0.0/8 cidr:172.0.0.0/9 cidr:172.16.0.0/12 cidr:172.16.0.0/13 cidr:172.16.0.0/14 cidr:172.18.0.0/15 cidr:172.18.0.0/16 cidr:172.18.0.0/17 cidr:172.18.0.0/18 cidr:172.18.0.0/19 cidr:172.18.0.0/20 cidr:172.18.0.0/21 cidr:172.18.0.0/22 cidr:172.18.0.0/23 cidr:172.18.0.0/24 cidr:172.18.0.0/25 cidr:172.18.0.0/26 cidr:172.18.0.0/27 cidr:172.18.0.0/28 cidr:172.18.0.0/29 cidr:172.18.0.4/30 cidr:172.18.0.4/31 cidr:172.18.0.4/32 reserved:kube-apiserver reserved:world]" subsys=policy
    testcase.go:279: lbmap mismatch:
        --- /home/travis/gopath/src/github.com/cilium/cilium/test/controlplane/services/nodeport/v1.25/lbmap1_nodeport-worker2.golden
        +++ <actual>
        @@ -1,25 +1,13 @@
        - - Services ----------------------------------------------------------------
        -| ID |               Name |      Type |              Frontend | Backend IDs |
        -|----+--------------------+-----------+-----------------------+-------------
        -|  0 | default/kubernetes | ClusterIP |    10.96.0.1:443/NONE |           1 |
        -|  1 |          test/echo | ClusterIP |  10.96.50.104:80/NONE |           0 |
        -|  2 |    test/echo-local | ClusterIP |  10.96.155.18:80/NONE |           0 |
        -|  3 |          test/echo |  NodePort |    0.0.0.0:30781/NONE |           0 |
        -|  4 |          test/echo |  NodePort |   10.0.0.2:30781/NONE |           0 |
        -|  5 |          test/echo |  NodePort |   10.0.0.3:30781/NONE |           0 |
        -|  6 |    test/echo-local |  NodePort |    0.0.0.0:30867/NONE |             |
        -|  7 |    test/echo-local |  NodePort |  0.0.0.0:30867/NONE/i |           0 |
        -|  8 |    test/echo-local |  NodePort |   10.0.0.2:30867/NONE |             |
        -|  9 |    test/echo-local |  NodePort | 10.0.0.2:30867/NONE/i |           0 |
        -| 10 |    test/echo-local |  NodePort |   10.0.0.3:30867/NONE |             |
        -| 11 |    test/echo-local |  NodePort | 10.0.0.3:30867/NONE/i |           0 |
        - ---------------------------------------------------------------------------
        + - Services -------------------------------------------------------------
        +| ID |               Name |      Type |           Frontend | Backend IDs |
        +|----+--------------------+-----------+--------------------+-------------
        +|  0 | default/kubernetes | ClusterIP | 10.96.0.1:443/NONE |           0 |
        + ------------------------------------------------------------------------
         
        - - Backends --------------------------------------------------
        -| ID |            L3n4Addr |  State |         Linked Services |
        -|----+---------------------+--------+-------------------------
        -|  0 | 10.244.1.224:80/TCP | active | 1, 2, 3, 4, 5, 7, 9, 11 |
        -|  1 | 172.18.0.4:6443/TCP | active |                       0 |
        - -------------------------------------------------------------
        + - Backends ------------------------------------------
        +| ID |            L3n4Addr |  State | Linked Services |
        +|----+---------------------+--------+-----------------
        +|  0 | 172.18.0.4:6443/TCP | active |               0 |
        + -----------------------------------------------------
           
```

Seen in https://github.com/cilium/cilium/pull/21506/checks?check_run_id=8815450765

Just a minor note that this happened in arm64-graviton2

@joamaki can you take an initial look?

--- a/test/controlplane/ciliumnetworkpolicies/status_updates_gc.go
+++ b/test/controlplane/ciliumnetworkpolicies/status_updates_gc.go
@@ -243,27 +243,7 @@ func init() {
 		// We only need to test the last k8s version
 		test := suite.NewControlPlaneTest(t, "cnp-status-update-control-plane", k8sVersions[len(k8sVersions)-1])
 
-		// When running with GC disabled, the Nodes Status updates should not be deleted.
-		test.
-			UpdateObjects(initialObjects...).
-			SetupEnvironment().
-			// check that CNPs contain status updates info before starting agent and operator
-			Eventually(func() error { return validateCNPs(test) }).
-			StartAgent(func(_ *option.DaemonConfig) {}).
-			StartOperator(
-				func(operatorCfg *operatorOption.OperatorConfig) {
-				},
-				func(vp *viper.Viper) {
-					vp.Set(operatorApi.OperatorAPIServeAddr, "localhost:0")
-				},
-			).
-			Eventually(func() error { return validateCNPs(test) }).
-			StopAgent().
-			StopOperator().
-			DeleteObjects(initialObjects...).
-			ClearEnvironment()
-
-		// When running with GC enabled, the Nodes Status updates should eventually be deleted.
+		// CNP nodes status updates are deprecated and should be deleted.
 		test.
 			UpdateObjects(initialObjects...).
 			SetupEnvironment().

--- a/test/controlplane/node/ciliumnodes/ciliumnodes.go
+++ b/test/controlplane/node/ciliumnodes/ciliumnodes.go
@@ -90,7 +90,9 @@ func init() {
 				test.
 					UpdateObjectsFromFile(abs("init.yaml")).
 					SetupEnvironment().
+					RecordWatchers().
 					StartAgent(modConfig).
+					EnsureWatchers("nodes").
 					ClearEnvironment()
 
 				// Run through the test steps

--- a/test/controlplane/pod/hostport/hostport.go
+++ b/test/controlplane/pod/hostport/hostport.go
@@ -36,7 +36,9 @@ func testHostPort(t *testing.T) {
 	test.
 		UpdateObjectsFromFile(abs("init.yaml")).
 		SetupEnvironment().
+		RecordWatchers().
 		StartAgent(func(_ *option.DaemonConfig) {}).
+		EnsureWatchers("pods").
 
 		// Step 1: Create the first hostport pod.
 		// lbmap1.golden: Hostport service exists in the Datapath with hostport-1 pod as backend.

--- a/test/controlplane/services/dualstack/dualstack.go
+++ b/test/controlplane/services/dualstack/dualstack.go
@@ -40,7 +40,9 @@ func testDualStack(t *testing.T) {
 			test.
 				UpdateObjectsFromFile(abs("init.yaml")).
 				SetupEnvironment().
+				RecordWatchers().
 				StartAgent(modConfig).
+				EnsureWatchers("endpointslices", "services").
 				UpdateObjectsFromFile(abs("state1.yaml")).
 				Eventually(func() error { return validate(abs("lbmap1.golden"), test) }).
 				StopAgent().

--- a/test/controlplane/services/graceful-termination/graceful_termination.go
+++ b/test/controlplane/services/graceful-termination/graceful_termination.go
@@ -40,7 +40,9 @@ func testGracefulTermination(t *testing.T) {
 	test.
 		UpdateObjectsFromFile(abs("init.yaml")).
 		SetupEnvironment().
+		RecordWatchers().
 		StartAgent(modConfig).
+		EnsureWatchers("endpointslices", "services").
 
 		// Step 1: Initial creation of the services and backends
 		// lbmap1.golden: Shows graceful-term-svc service with an active backend

--- a/test/controlplane/services/nodeport/nodeport.go
+++ b/test/controlplane/services/nodeport/nodeport.go
@@ -40,7 +40,9 @@ func init() {
 					test.
 						UpdateObjectsFromFile(abs("init.yaml")).
 						SetupEnvironment().
+						RecordWatchers().
 						StartAgent(modConfig).
+						EnsureWatchers("endpointslices", "pods", "services").
 						UpdateObjectsFromFile(abs("state1.yaml")).
 						Eventually(func() error { return validate(test, abs("lbmap1_"+nodeName+".golden")) }).
 						StopAgent().

--- a/test/controlplane/suite/testcase.go
+++ b/test/controlplane/suite/testcase.go
@@ -469,8 +469,6 @@ var _ watch.Interface = &filteringWatcher{}
 
 func (fw *filteringWatcher) Stop() {
 	fw.parent.Stop()
-	close(fw.events)
-	fw.events = nil
 }
 
 func (fw *filteringWatcher) ResultChan() <-chan watch.Event {
@@ -486,6 +484,7 @@ func (fw *filteringWatcher) ResultChan() <-chan watch.Event {
 				fw.events <- event
 			}
 		}
+		close(fw.events)
 	}()
 	return fw.events
 }

--- a/test/controlplane/suite/testcase.go
+++ b/test/controlplane/suite/testcase.go
@@ -36,6 +36,7 @@ import (
 	k8sClient "github.com/cilium/cilium/pkg/k8s/client"
 	slim_corev1 "github.com/cilium/cilium/pkg/k8s/slim/k8s/api/core/v1"
 	"github.com/cilium/cilium/pkg/k8s/version"
+	"github.com/cilium/cilium/pkg/lock"
 	"github.com/cilium/cilium/pkg/node/types"
 	agentOption "github.com/cilium/cilium/pkg/option"
 )
@@ -50,12 +51,13 @@ type ControlPlaneTest struct {
 	tempDir           string
 	validationTimeout time.Duration
 
-	nodeName       string
-	clients        *k8sClient.FakeClientset
-	trackers       []trackerAndDecoder
-	agentHandle    *agentHandle
-	operatorHandle *operatorHandle
-	Datapath       *fakeTypes.FakeDatapath
+	nodeName            string
+	clients             *k8sClient.FakeClientset
+	trackers            []trackerAndDecoder
+	agentHandle         *agentHandle
+	operatorHandle      *operatorHandle
+	Datapath            *fakeTypes.FakeDatapath
+	establishedWatchers lock.Map[string, struct{}]
 }
 
 func NewControlPlaneTest(t *testing.T, nodeName string, k8sVersion string) *ControlPlaneTest {
@@ -254,6 +256,48 @@ func (cpt *ControlPlaneTest) Get(gvr schema.GroupVersionResource, ns, name strin
 		}
 	}
 	return nil, err
+}
+
+// RecordWatchers intercepts 'Watch' calls on the fake k8s clientsets. The reason we need to do this
+// is the following: The k8s object tracker's implementation of Watch is not equivalent to Watch on
+// a real api-server, as it does not respect the ResourceVersion from whence to start the watch. As
+// a consequence, when informers (or reflectors) call ListAndWatch, they miss events which occur
+// between the end of List and the establishment of Watch.
+//
+// To decrease the likelihood of this race occurring in the control plane tests, we install a
+// mechanism to wait for watchers of specific resources: see also EnsureWatchers. This isn't a
+// complete fix - if multiple watchers for the same resource are established, this may give false
+// positives.
+func (cpt *ControlPlaneTest) RecordWatchers() *ControlPlaneTest {
+	reaction := func(action k8sTesting.Action) (handled bool, ret watch.Interface, err error) {
+		r := action.GetResource().Resource
+		if _, ok := cpt.establishedWatchers.Load(r); ok {
+			cpt.t.Logf("WARNING: Multiple watches for resource %s intercepted. This highlights a potential cause for flakes", r)
+		}
+		cpt.establishedWatchers.Store(r, struct{}{})
+		return false, nil, nil
+	}
+
+	cpt.clients.SlimFakeClientset.PrependWatchReactor("*", reaction)
+	cpt.clients.KubernetesFakeClientset.PrependWatchReactor("*", reaction)
+	cpt.clients.CiliumFakeClientset.PrependWatchReactor("*", reaction)
+	cpt.clients.APIExtFakeClientset.PrependWatchReactor("*", reaction)
+	return cpt
+}
+
+// EnsureWatchers delays progress of the test until watchers for resources have been established on
+// the clientset.
+func (cpt *ControlPlaneTest) EnsureWatchers(resources ...string) *ControlPlaneTest {
+	cpt.retry(func() error {
+		for _, resource := range resources {
+			if _, ok := cpt.establishedWatchers.Load(resource); !ok {
+				return fmt.Errorf("no watcher for %s yet", resource)
+			}
+		}
+		return nil
+	})
+
+	return cpt
 }
 
 func (cpt *ControlPlaneTest) UpdateObjectsFromFile(filename string) *ControlPlaneTest {

